File: tauri/src-tauri/src/core/prompt/identity.rs
use super::PromptPart;

pub struct IdentityPart {
    pub user_name: String,
}

impl PromptPart for IdentityPart {
    fn render(&self) -> String {
        format!(
            "You are MOSAIC, a highly capable AI coding agent. You operate inside a workspace \
             and have direct access to tools for reading, writing, editing files, running commands, \
             and searching code.\n\
             Your user is {}. Be concise, precise, and action-oriented.",
            self.user_name
        )
    }
}

File: electron/electron/core/prompt/identity.ts
import type { PromptPart } from './types'

class IdentityPart implements PromptPart {
    private userName: string

    constructor(userName: string) {
        this.userName = userName
    }

    render(): string {
        return `You are MOSAIC, a highly capable AI coding agent. You operate inside a workspace and have direct access to tools for reading, writing, editing files, running commands, and searching code.\nYour user is ${this.userName}. Be concise, precise, and action-oriented.`
    }
}

export default IdentityPart
